integrationTests/ts/__benchmarks__/benchmarks.ts

```diff
@@ -0,0 +1,367 @@
+
+        const ecdhKeypair = new Keypair();
+        const sharedKey = Keypair.genEcdhSharedKey(ecdhKeypair.privKey, coordinatorKeypair.pubKey);
+        const message = command.encrypt(signature, sharedKey);
+        poll?.publishMessage(message, ecdhKeypair.pubKey);
+
+        while (poll?.hasUnprocessedMessages()) {
+          const pmInputs = poll.processMessages(pollId) as unknown as CircuitInputs;
+
+          // eslint-disable-next-line no-await-in-loop
+          await genProof({
+            inputs: pmInputs,
+            useWasm: true,
+            wasmPath: path.resolve(
+              __dirname,
+              "../../../cli/zkeys/ProcessMessages_10-2-1-2_test/ProcessMessages_10-2-1-2_test_js/ProcessMessages_10-2-1-2_test.wasm",
+            ),
+            zkeyPath: path.resolve(
+              __dirname,
+              "../../../cli/zkeys/ProcessMessages_10-2-1-2_test/ProcessMessages_10-2-1-2_test.0.zkey",
+            ),
+          });
+        }
+      }
+    }),
+
+    benny.add(`rapidnsark genProof - 1 signups, 5 messages`, async () => {
+      const maciState = new MaciState(STATE_TREE_DEPTH);
+      const pollId = maciState.deployPoll(
+        BigInt(duration),
+        maxValues,
+        treeDepths,
+        messageBatchSize,
+        coordinatorKeypair,
+      );
+
+      const poll = maciState.polls.get(pollId);
+
+      maciState.signUp(users[0].pubKey, voiceCreditBalance, BigInt(Date.now()));
+
+      // submit 5 messages
+      // eslint-disable-next-line @typescript-eslint/prefer-for-of
+      for (let i = 0; i < users.length; i += 1) {
+        const keypair = users[i];
+        // First command (valid)
+        const command = new PCommand(
+          BigInt(i) + 1n,
+          keypair.pubKey,
+          5n, // voteOptionIndex,
+          7n, // vote weight
+          2n, // nonce
+          pollId,
+        );
+
+        const signature = command.sign(keypair.privKey);
+
+        const ecdhKeypair = new Keypair();
+        const sharedKey = Keypair.genEcdhSharedKey(ecdhKeypair.privKey, coordinatorKeypair.pubKey);
+        const message = command.encrypt(signature, sharedKey);
+        poll?.publishMessage(message, ecdhKeypair.pubKey);
+
+        while (poll?.hasUnprocessedMessages()) {
+          const pmInputs = poll.processMessages(pollId) as unknown as CircuitInputs;
+
+          // eslint-disable-next-line no-await-in-loop
+          await genProof({
+            inputs: pmInputs,
+            useWasm: true,
+            rapidsnarkExePath: `${homedir()}/rapidsnark/build/prover`,
+            witnessExePath: path.resolve(
+              __dirname,
+              "../../../cli/zkeys/ProcessMessages_10-2-1-2_test/ProcessMessages_10-2-1-2_test_cpp/ProcessMessages_10-2-1-2_test",
+            ),
+            zkeyPath: path.resolve(
+              __dirname,
+              "../../../cli/zkeys/ProcessMessages_10-2-1-2_test/ProcessMessages_10-2-1-2_test.0.zkey",
+            ),
+          });
```

Comment:
Is this expected that this loop is nested?

Reply:
nope, moved it out